### ISSUE TITLE
Share cannot start if initial members not set

### ISSUE THREAD

**Issue by andreaskring** [aldica/aldica#57]
## BUG

An aldica-enabled Share container cannot start without errors if `aldica.core.initialMembers` is not set.

### Expected behavior

Share should be able to start / join grid even if `aldica.core.initialMembers` is not set.

### Actual behavior

When the Share container(s) start up, the error below is thrown in the log.

### Steps to reproduce the behavior

Remove the two lines containing the `aldica.core.initialMembers` setting from the Share containers in our `docker-compose.yml` file and start the stack with `docker-compose up -d --build`.

### Additional details (analysis so far, log statements, references, etc.)

The following error is thrown in the log:
```
share1_1                 |  [14:39:42]    __________  ________________ 
share1_1                 | [14:39:42]   /  _/ ___/ |/ /  _/_  __/ __/ 
share1_1                 | [14:39:42]  _/ // (7 7    // /  / / / _/   
share1_1                 | [14:39:42] /___/\___/_/|_/___/ /_/ /___/  
share1_1                 | [14:39:42] 
share1_1                 | [14:39:42] ver. 2.8.1#20200521-sha1:86422096
share1_1                 | [14:39:42] 2020 Copyright(C) Apache Software Foundation
share1_1                 | [14:39:42] 
share1_1                 | [14:39:42] Ignite documentation: http://ignite.apache.org
share1_1                 | [14:39:42] 
share1_1                 | [14:39:42] Quiet mode.
share1_1                 | [14:39:42]   ^-- Logging by 'Slf4jLogger [impl=org.slf4j.impl.Log4jLoggerAdapter(o.a.i.i.IgniteKernal%shareGrid), quiet=true]'
share1_1                 | [14:39:42]   ^-- To see **FULL** console log here add -DIGNITE_QUIET=false or "-v" to ignite.{sh|bat}
share1_1                 | [14:39:42] 
share1_1                 | [14:39:42] OS: Linux 4.15.0-122-generic amd64
share1_1                 | [14:39:42] VM information: OpenJDK Runtime Environment 11.0.7+10 Oracle Corporation OpenJDK 64-Bit Server VM 11.0.7+10
share1_1                 | 2020-11-13 14:39:42,051  WARN  [ignite.internal.IgniteKernal%shareGrid] [localhost-startStop-1] Please set system property '-Djava.net.preferIPv4Stack=true' to avoid possible problems in mixed environments.
share1_1                 |  [14:39:42] Please set system property '-Djava.net.preferIPv4Stack=true' to avoid possible problems in mixed environments.
share1_1                 | 2020-11-13 14:39:42,144  WARN  [ignite.internal.GridDiagnostic] [pub-#12%shareGrid%] Initial heap size is 500MB (should be no less than 512MB, use -Xms512m -Xmx512m).
share1_1                 |  [14:39:42] Initial heap size is 500MB (should be no less than 512MB, use -Xms512m -Xmx512m).
share1_1                 | [14:39:42] Configured plugins:
share1_1                 | [14:39:42]   ^-- SimpleSecurityPlugin (org.aldica:aldica-common-ignite) 1.0.0
share1_1                 | [14:39:42]   ^-- Copyright 2020 Acosix GmbH, Copyright 2020 MAGENTA ApS
share1_1                 | [14:39:42] 
share1_1                 | [14:39:42] Configured failure handler: [hnd=StopNodeOrHaltFailureHandler [tryStop=false, timeout=0, super=AbstractFailureHandler [ignoredFailureTypes=UnmodifiableSet [SYSTEM_WORKER_BLOCKED, SYSTEM_CRITICAL_OPERATION_TIMEOUT]]]]
share1_1                 | 2020-11-13 14:39:42,373  INFO  [ignite.plugin.SimpleSecurityProcessor] [localhost-startStop-1] No allowed client credentials have been configured - no client will be allowed to connect
share1_1                 |  2020-11-13 14:39:42,764  WARN  [communication.tcp.TcpCommunicationSpi] [localhost-startStop-1] Failure detection timeout will be ignored (one of SPI parameters has been set explicitly)
share1_1                 |  [14:39:43] Security status [authentication=on, tls/ssl=off]
share1_1                 | 2020-11-13 14:39:43,356  WARN  [ignite.discovery.CredentialsAwareTcpDiscoverySpi] [localhost-startStop-1] Failure detection timeout will be ignored (one of SPI parameters has been set explicitly)
share1_1                 |  2020-11-13 14:39:43,416  WARN  [cache.persistence.IgniteCacheDatabaseSharedManager] [localhost-startStop-1] DataRegionConfiguration.maxWalArchiveSize instead DataRegionConfiguration.walHistorySize would be used for removing old archive wal files
share1_1                 |  2020-11-13 14:39:43,453  WARN  [dht.topology.PartitionsEvictManager] [localhost-startStop-1] Logging at INFO level without checking if INFO level is enabled: Evict partition permits=1
share1_1                 |  2020-11-13 14:39:43,987  INFO  [ignite.discovery.CredentialsAwareTcpDiscoverySpi] [localhost-startStop-1] Connection check threshold is calculated: 2000
share1_1                 |  2020-11-13 14:39:44,014  INFO  [ignite.discovery.CredentialsAwareTcpDiscoverySpi] [localhost-startStop-1] Successfully bound to TCP port [port=47140, localHost=share1/172.24.0.8, locNodeId=25af04d7-bdbf-4529-ad41-b2ccf1188750]
share1_1                 |  2020-11-13 14:39:44,028  ERROR [ignite.internal.IgniteKernal%shareGrid] [localhost-startStop-1] Failed to start manager: GridManagerAdapter [enabled=true, name=o.a.i.i.managers.discovery.GridDiscoveryManager]
share1_1                 |  class org.apache.ignite.IgniteCheckedException: Failed to start SPI: TcpDiscoverySpi [addrRslvr=org.aldica.common.ignite.discovery.GridAddressResolutionManager@45d59dcc, sockTimeout=5000, ackTimeout=5000, marsh=JdkMarshaller [clsFilter=org.apache.ignite.marshaller.MarshallerUtils$1@6674f891], reconCnt=10, reconDelay=2000, maxAckTimeout=600000, soLinger=5, forceSrvMode=false, clientReconnectDisabled=false, internalLsnr=null, skipAddrsRandomization=false]
share1_1                 | 	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:302)
share1_1                 | 	at org.apache.ignite.internal.managers.discovery.GridDiscoveryManager.start(GridDiscoveryManager.java:943)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.startManager(IgniteKernal.java:1960)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.start(IgniteKernal.java:1276)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start0(IgnitionEx.java:2045)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start(IgnitionEx.java:1703)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start0(IgnitionEx.java:1117)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:637)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:563)
share1_1                 | 	at org.apache.ignite.Ignition.start(Ignition.java:321)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.lambda$initializeGrid$2(SpringIgniteLifecycleBean.java:130)
share1_1                 | 	at org.aldica.common.ignite.context.ExternalContext.withExternalContext(ExternalContext.java:112)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.initializeGrid(SpringIgniteLifecycleBean.java:120)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.onApplicationEvent(SpringIgniteLifecycleBean.java:68)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:896)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:552)
share1_1                 | 	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:401)
share1_1                 | 	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:292)
share1_1                 | 	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103)
share1_1                 | 	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4699)
share1_1                 | 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5165)
share1_1                 | 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:743)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:719)
share1_1                 | 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:714)
share1_1                 | 	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1125)
share1_1                 | 	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1859)
share1_1                 | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
share1_1                 | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
share1_1                 | 	at java.base/java.lang.Thread.run(Thread.java:834)
share1_1                 | Caused by: class org.apache.ignite.spi.IgniteSpiException: Non-shared IP finder must have IP addresses specified in TcpDiscoveryIpFinder.getRegisteredAddresses() configuration property (specify list of IP addresses in configuration).
share1_1                 | 	at org.apache.ignite.spi.discovery.tcp.ServerImpl.spiStart(ServerImpl.java:413)
share1_1                 | 	at org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi.spiStart(TcpDiscoverySpi.java:2099)
share1_1                 | 	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:299)
share1_1                 | 	... 36 more
share1_1                 | 2020-11-13 14:39:44,044  ERROR [ignite.internal.IgniteKernal%shareGrid] [localhost-startStop-1] Got exception while starting (will rollback startup routine).
share1_1                 |  class org.apache.ignite.IgniteCheckedException: Failed to start manager: GridManagerAdapter [enabled=true, name=org.apache.ignite.internal.managers.discovery.GridDiscoveryManager]
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.startManager(IgniteKernal.java:1965)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.start(IgniteKernal.java:1276)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start0(IgnitionEx.java:2045)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start(IgnitionEx.java:1703)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start0(IgnitionEx.java:1117)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:637)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:563)
share1_1                 | 	at org.apache.ignite.Ignition.start(Ignition.java:321)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.lambda$initializeGrid$2(SpringIgniteLifecycleBean.java:130)
share1_1                 | 	at org.aldica.common.ignite.context.ExternalContext.withExternalContext(ExternalContext.java:112)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.initializeGrid(SpringIgniteLifecycleBean.java:120)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.onApplicationEvent(SpringIgniteLifecycleBean.java:68)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:896)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:552)
share1_1                 | 	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:401)
share1_1                 | 	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:292)
share1_1                 | 	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103)
share1_1                 | 	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4699)
share1_1                 | 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5165)
share1_1                 | 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:743)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:719)
share1_1                 | 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:714)
share1_1                 | 	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1125)
share1_1                 | 	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1859)
share1_1                 | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
share1_1                 | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
share1_1                 | 	at java.base/java.lang.Thread.run(Thread.java:834)
share1_1                 | Caused by: class org.apache.ignite.IgniteCheckedException: Failed to start SPI: TcpDiscoverySpi [addrRslvr=org.aldica.common.ignite.discovery.GridAddressResolutionManager@45d59dcc, sockTimeout=5000, ackTimeout=5000, marsh=JdkMarshaller [clsFilter=org.apache.ignite.marshaller.MarshallerUtils$1@6674f891], reconCnt=10, reconDelay=2000, maxAckTimeout=600000, soLinger=5, forceSrvMode=false, clientReconnectDisabled=false, internalLsnr=null, skipAddrsRandomization=false]
share1_1                 | 	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:302)
share1_1                 | 	at org.apache.ignite.internal.managers.discovery.GridDiscoveryManager.start(GridDiscoveryManager.java:943)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.startManager(IgniteKernal.java:1960)
share1_1                 | 	... 34 more
share1_1                 | Caused by: class org.apache.ignite.spi.IgniteSpiException: Non-shared IP finder must have IP addresses specified in TcpDiscoveryIpFinder.getRegisteredAddresses() configuration property (specify list of IP addresses in configuration).
share1_1                 | 	at org.apache.ignite.spi.discovery.tcp.ServerImpl.spiStart(ServerImpl.java:413)
share1_1                 | 	at org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi.spiStart(TcpDiscoverySpi.java:2099)
share1_1                 | 	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:299)
share1_1                 | 	... 36 more
share1_1                 | 2020-11-13 14:39:59,097  INFO  [ignite.discovery.CredentialsAwareTcpDiscoverySpi] [localhost-startStop-1] No verification for local node leave has been received from coordinator (will stop node anyway).
share1_1                 |  2020-11-13 14:39:59,100  ERROR [ignite.internal.IgniteKernal%shareGrid] [localhost-startStop-1] Failed to stop component (ignoring): GridProcessorAdapter []
share1_1                 |  java.lang.UnsupportedOperationException
share1_1                 | 	at org.jsr166.ConcurrentLinkedHashMap.clear(ConcurrentLinkedHashMap.java:1542)
share1_1                 | 	at org.apache.ignite.internal.processors.job.GridJobProcessor.stop(GridJobProcessor.java:375)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.stop0(IgniteKernal.java:2627)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.stop(IgniteKernal.java:2499)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.start(IgniteKernal.java:1395)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start0(IgnitionEx.java:2045)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start(IgnitionEx.java:1703)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start0(IgnitionEx.java:1117)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:637)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:563)
share1_1                 | 	at org.apache.ignite.Ignition.start(Ignition.java:321)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.lambda$initializeGrid$2(SpringIgniteLifecycleBean.java:130)
share1_1                 | 	at org.aldica.common.ignite.context.ExternalContext.withExternalContext(ExternalContext.java:112)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.initializeGrid(SpringIgniteLifecycleBean.java:120)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.onApplicationEvent(SpringIgniteLifecycleBean.java:68)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:896)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:552)
share1_1                 | 	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:401)
share1_1                 | 	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:292)
share1_1                 | 	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103)
share1_1                 | 	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4699)
share1_1                 | 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5165)
share1_1                 | 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:743)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:719)
share1_1                 | 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:714)
share1_1                 | 	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1125)
share1_1                 | 	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1859)
share1_1                 | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
share1_1                 | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
share1_1                 | 	at java.base/java.lang.Thread.run(Thread.java:834)
share1_1                 | [14:39:59] Ignite node stopped wih ERRORS [name=shareGrid, uptime=00:00:24.224]
share1_1                 | 2020-11-13 14:39:59,126  ERROR [web.context.ContextLoader] [localhost-startStop-1] Context initialization failed
share1_1                 |  class org.apache.ignite.IgniteException: Failed to start manager: GridManagerAdapter [enabled=true, name=org.apache.ignite.internal.managers.discovery.GridDiscoveryManager]
share1_1                 | 	at org.apache.ignite.internal.util.IgniteUtils.convertException(IgniteUtils.java:1067)
share1_1                 | 	at org.apache.ignite.Ignition.start(Ignition.java:324)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.lambda$initializeGrid$2(SpringIgniteLifecycleBean.java:130)
share1_1                 | 	at org.aldica.common.ignite.context.ExternalContext.withExternalContext(ExternalContext.java:112)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.initializeGrid(SpringIgniteLifecycleBean.java:120)
share1_1                 | 	at org.aldica.common.ignite.lifecycle.SpringIgniteLifecycleBean.onApplicationEvent(SpringIgniteLifecycleBean.java:68)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
share1_1                 | 	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:896)
share1_1                 | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:552)
share1_1                 | 	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:401)
share1_1                 | 	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:292)
share1_1                 | 	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103)
share1_1                 | 	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4699)
share1_1                 | 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5165)
share1_1                 | 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:743)
share1_1                 | 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:719)
share1_1                 | 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:714)
share1_1                 | 	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1125)
share1_1                 | 	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1859)
share1_1                 | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
share1_1                 | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
share1_1                 | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
share1_1                 | 	at java.base/java.lang.Thread.run(Thread.java:834)
share1_1                 | Caused by: class org.apache.ignite.IgniteCheckedException: Failed to start manager: GridManagerAdapter [enabled=true, name=org.apache.ignite.internal.managers.discovery.GridDiscoveryManager]
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.startManager(IgniteKernal.java:1965)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.start(IgniteKernal.java:1276)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start0(IgnitionEx.java:2045)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start(IgnitionEx.java:1703)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start0(IgnitionEx.java:1117)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:637)
share1_1                 | 	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:563)
share1_1                 | 	at org.apache.ignite.Ignition.start(Ignition.java:321)
share1_1                 | 	... 27 more
share1_1                 | Caused by: class org.apache.ignite.IgniteCheckedException: Failed to start SPI: TcpDiscoverySpi [addrRslvr=org.aldica.common.ignite.discovery.GridAddressResolutionManager@45d59dcc, sockTimeout=5000, ackTimeout=5000, marsh=JdkMarshaller [clsFilter=org.apache.ignite.marshaller.MarshallerUtils$1@6674f891], reconCnt=10, reconDelay=2000, maxAckTimeout=600000, soLinger=5, forceSrvMode=false, clientReconnectDisabled=false, internalLsnr=null, skipAddrsRandomization=false]
share1_1                 | 	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:302)
share1_1                 | 	at org.apache.ignite.internal.managers.discovery.GridDiscoveryManager.start(GridDiscoveryManager.java:943)
share1_1                 | 	at org.apache.ignite.internal.IgniteKernal.startManager(IgniteKernal.java:1960)
share1_1                 | 	... 34 more
share1_1                 | Caused by: class org.apache.ignite.spi.IgniteSpiException: Non-shared IP finder must have IP addresses specified in TcpDiscoveryIpFinder.getRegisteredAddresses() configuration property (specify list of IP addresses in configuration).
share1_1                 | 	at org.apache.ignite.spi.discovery.tcp.ServerImpl.spiStart(ServerImpl.java:413)
share1_1                 | 	at org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi.spiStart(TcpDiscoverySpi.java:2099)
share1_1                 | 	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:299)
share1_1                 | 	... 36 more
share1_1                 | 13-Nov-2020 14:39:59.378 SEVERE [localhost-startStop-1] org.apache.catalina.core.StandardContext.startInternal One or more listeners failed to start. Full details will be found in the appropriate container log file
```

### Tell us about your environment
Docker Compose file (modified as described above) provided by the aldica project.

**Comment by AFaust** [comment]
Initial members are required for Share - otherwise the module won't ever find another server to connect to and would be pointless. But I realise our documentation contains a copy&paste error in the (Share config reference)[https://github.com/aldica/aldica/blob/master/docs/Configuration-ShareReference.md#network-properties] where the description is Repository-specific.

**Comment by AFaust** [comment]
Also see the (simple Share scenario)[https://github.com/aldica/aldica/blob/master/docs/GettingStarted-Share-Scenarios.md#simple-grid-with-explicitly-configured-initial-members]...

**Comment by andreaskring** [comment]
Yes - right... I remember now - thanks for the info. I looked at the documentation error where is says: "...This is generally not necessary as aldica will handle discovery using database-stored address registrations.". The setting of the initial members flag makes deployment with Kubernetes a bit more difficult since the Share pods need to be aware of each other. I will make a small PR that addresses the copy&paste error in the documentation.

**Comment by AFaust** [comment]
I wonder if we should also provide a "Share Companion App" - but other than doing the cache extension like in Repository tier, its purpose would be nothing more than being a central registry for the Share grid. That would allow to configure the Share pods with a single, static initial member using host / service name. Share instances would connect to it, and get address information from other Share instances that have already connected. In order to support failure / reconnect scenarios, the companion app might have to write some kind of persistent state to a persistent volume claim, as otherwise it won't be able to re-connect to an existing grid when it comes back after a failure.

**Comment by andreaskring** [comment]
@AFaust Sorry - must have missed this - just discovered your update now... Yes - it would be great if we had a "Share Companion App" that - apart from being a cache extension - could provide the Share grid with an initial member. Regarding your PVC suggestion - we actually ended up implementing something like this in our Helm chart (a fork of the upstream acs-deployment Helm chart which will be placed on GitHub and then we can link to it from the aldica-project (or we can place it under the aldica GitHub organisation)). We do the following:
* Create a small PVC for Share (common to all Share pods)
* Add a script to the `ENTRYPOINT` in the `Dockerfile` for Share that will 1) add the IP address of the Share pod to a file which is stored on the common PV for Share and 2) set the `aldica.core.initialMembers` to the full list of all pods in the grid (which will only be a single pod when the first Share pod start)
* Add a `preStop` command to the `lifecycle` hook in the Share deployment YAML (which should probably be changed to a `StatefulSet`) that will remove the pod IP from the list persisted in the common PV when pod is terminated.

So basically we maintain a simple file that contains the Share pods in the Ignite grid and use this to set the initial members
